src/metabase/query_processor/middleware/upgrade_field_literals.clj

```diff
@@ -26,32 +20,15 @@
     (qp.store/cached ::bad-clause-warning
       (log/warn (u/colorize :red message)))))
 
-(defn- fix-clause [{:keys [inner-query source-aliases field-name->field]} [_ field-name options :as field-clause]]
+(defn- fix-clause [{:keys [source-aliases field-name->field]} [_ field-name options :as field-clause]]
   ;; attempt to find a corresponding Field ref from the source metadata.
   (let [field-ref (:field_ref (get field-name->field field-name))]
     (cond
       field-ref
       (mbql.u/match-one field-ref
-        ;; If the matching Field ref is an integer `:field` clause then replace it with the corrected clause and log
-        ;; a developer-facing warning. Things will still work and this should be fixed on the FE, but we don't need to
-        ;; blow up prod logs
+        ;; If the matching Field ref is an integer `:field` clause then replace it with the corrected clause.
         [:field (id :guard integer?) new-options]
-        (u/prog1 [:field id (merge new-options (dissoc options :base-type))]
-          (when (and (not config/is-prod?)
-                     (not (has-a-native-source-query-at-some-level? inner-query)))
-            ;; don't i18n this because it's developer-facing only.
-            (warn-once
-             (str "Warning: query is using a [:field <string> ...] clause to refer to a Field in an MBQL source query."
-                  \newline
-                  "Use [:field <integer> ...] clauses to refer to Fields in MBQL source queries."
-                  \newline
-                  "We will attempt to fix this, but it may lead to incorrect queries."
-                  \newline
-                  "See #19757 for more information."
-                  \newline
-                  (str "Clause:       " (pr-str field-clause))
-                  \newline
-                  (str "Corrected to: " (pr-str <>))))))
+        [:field id (merge new-options (dissoc options :base-type))]
```

Comment:
This is completely unrelated, but since we're generating these queries now we should stop putting warnings like this in the logs, especially since we actually want to start moving towards refs that use `[:field {} <string>]` in stages after the first (see #29763)